src/util/expr.h

```diff
@@ -172,6 +173,10 @@ class exprt:public irept
   const_unique_depth_iteratort unique_depth_end() const;
   const_unique_depth_iteratort unique_depth_cbegin() const;
   const_unique_depth_iteratort unique_depth_cend() const;
+
+  //! Generates a string in a generic syntax
+  //! that is inspired by C/C++/Java, and is meant for debugging
+  std::string as_string() const;
```

Comment:
May I suggest not to make this a member function? The current implementation would introduce the same problem that jsont has: It makes it impossible to use a different output function, e.g., for generating some XML-formatted expression trees.

Reply:
I.e., just a function, say std::string as_string(const exprt &)?

Reply:
That was my thinking, yes. Upon further deliberation it might be worthwhile putting all of them in a separate class, which could then truly become a visitor. Either way will disentangle data representation from the output.